tests/pass/libc.rs

```diff
@@ -1,17 +1,147 @@
 // ignore-windows: No libc on Windows
 // compile-flags: -Zmiri-disable-isolation
 
+#![feature(io_error_more)]
 #![feature(rustc_private)]
 
+use std::path::PathBuf;
+
 extern crate libc;
 
-#[cfg(any(target_os = "linux", target_os = "freebsd"))]
-fn tmp() -> std::path::PathBuf {
+fn tmp() -> PathBuf {
     std::env::var("MIRI_TEMP")
-        .map(std::path::PathBuf::from)
+        .map(|tmp| {
+            // MIRI_TEMP is set outside of our emulated
+            // program, so it may have path separators that don't
+            // correspond to our target platform. We normalize them here
+            // before constructing a `PathBuf`
+
+            #[cfg(windows)]
+            return PathBuf::from(tmp.replace("/", "\\"));
+
+            #[cfg(not(windows))]
+            return PathBuf::from(tmp.replace("\\", "/"));
```

Comment:
```suggestion
            // before constructing a `PathBuf`.
            return PathBuf::from(tmp.replace("\\", "/"));
```
This is a non-windows test.